clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3683,10 +3686,12 @@ void Sema::CheckSYCLKernelCall(FunctionDecl *KernelFunc, SourceRange CallLoc,
   // issues.
   if (!LangOpts.SYCLIsDevice)
     return;
-  const CXXRecordDecl *KernelObj = getKernelObjectType(KernelFunc);
+  QualType KernelObjType = GetSYCLKernelObjectType(KernelFunc);
   QualType KernelNameType =
       calculateKernelNameType(getASTContext(), KernelFunc);
-  if (!KernelObj) {
+  const CXXRecordDecl *KernelObj;
+  if (KernelObjType.isNull() ||
+      !(KernelObj = KernelObjType->getAsCXXRecordDecl())) {
```

Comment:
```suggestion
  const CXXRecordDecl *KernelObj = !KernelObjType.isNull() ? KernelObjType->getAsCXXRecordDecl()) : nullptr;  
  if (!KernelObj) {
```
Though, of more interest to me is: can that `QualType` ever actually *be* null? I would assume that if it's null, that means there was some sort of error we already reported, but then why are we getting to *here* and giving a second error?

Reply:
I don't have a good idea of when it would happen actually... I couldn't come up with any examples, so i think it does make sense to just not have this condition and count on the `operator->` to cause the assert until we find it.  A non-asserts build will still emit the diagnostic though.